lib/search-bar/index.js

```diff
@@ -30,11 +32,19 @@ export default class SearchBarContainer extends Component {
   }
 
   handleInputChange (e) {
+    const input = e.target.value;
+    clearTimeout(keyPressTimeout);
+    keyPressTimeout = setTimeout(() => {
+      const analyticsObject = {
+        event: 'searchTyping',
+        searchInput: input
+      };
+      dataLayer.push(analyticsObject);
+    }, 3000);
```

Comment:
It only triggers after 3 secs if there is no another keypressed within this time.